security/advancedtls/crl.go

```diff
@@ -319,7 +183,7 @@ func fetchCRL(c *x509.Certificate, crlVerifyCrt []*x509.Certificate, cfg Revocat
 		}
 		return crl, nil
 	}
-	return fetchIssuerCRL(c.RawIssuer, crlVerifyCrt, cfg)
+	return nil, fmt.Errorf("trying to fetch CRL but CRLProvider is nil")
```

Comment:
Should we move this check to become the fist thing in this function?

Reply:
Good call, changed